todolist.js

```diff
@@ -1,97 +1,69 @@
-/**
- * 
- * Start of objects definition
- */
-
-
-/**
- * @returns {Object} Todos 
- * @param {Array<Todo>} todos optional array of todo objects
- */
-function Todos(todos = []) {
-    this.todos = todos;
+class Todo {
+    constructor(taskName, endDate, completed, description) {
+        this.taskName = taskName;
+        this.endDate = endDate;
+        this.completed = false;
+        this.description = description;
+    }
+    finish() {
+        this.completed = true;
+    }
+    modify(taskName, endDate, description) {
+        this.taskName = taskName;
+        this.endDate = endDate;
+        this.description = description;
+    }
 }
-
-/**
- * @param {Todo} todo
- */
-Todos.prototype.add = function(todo) {
-    this.todos.push(todo);
+class TodoList {
+    constructor() {
+        this.allTodos = new Array;
+    }
+    // lista de task-uri
+    listTodos() {
+        console.table(this.allTodos);
+    }
+    // functie de adaugare
+    newTodo(taskName, endDate, description) {
+        let addItem = new Todo(taskName, endDate, false, description);
+        this.allTodos.push(addItem);
+    }
+    // functie de stergere
+    delTask(index) {
+        this.allTodos.splice(index, 1);
+        this.listTodos();
+    }
+    // functie de modificare
+    modTask(index, taskName, endDate, description) {
+        this.allTodos[index].taskName = taskName || this.allTodos[index].taskName;
+        this.allTodos[index].endDate = endDate || this.allTodos[index].endDate;
+        this.allTodos[index].description = description || this.allTodos[index].description;
+        this.listTodos();
+    }
+    // functie pentru finalizarea unui task
+    endTask(index) {
+        this.allTodos[index].finish();
+        this.listTodos();
+    }
+    // functie listare taskuri nefinalizate
+    listActiveTodos() {
+        const finished = this.allTodos.filter(t => !t.completed);
+        console.table(finished);
+        this.listTodos();
```

Comment:
Nu mai este nevoie de `this.listTodos()` pentru ca metoda ta listeaza toate task-urile active, deci nu ma astept sa vad toate task-urile active iar apoi toate task-urile.
Best practice: O metoda ar trebui sa faca un singur lucru, si sa il faca bine.

Reply:
Multumesc pentru observatii si recomandari, `new Array()` a fost o scapare nefericita, trebuie sa fiu mai atent pe viitor.

La ultima metoda am adaugat `this.listTodos()` pentru ca in cerinte se specifica ca la toate punctele intre 4-8 sa listam toate task-urile, iar acest punct era singurul unde raspunsul la tema diferea de lista tuturor task-urilor, dar oricum apreciez sugestia de best practice :)

Reply:
Ok, inteleg, ideea mea era sa se listeze toate task-urile dupa ce se modifica array-ul, ca sa vedem corectitudinea modificarilor. :)